### DISCUSSION TITLE
Rpi2040 file write a bit iffy

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#9736] [Using MicroPython]
I've got a little config file I try to write before running boot.py during testing. It seems to work OK
```
>>> import json, time; f=open('config','w'); json.dump({'tests':7, 'dum':01, 'web':1, 'oledd':0, 'dormant':1}, f); time.sleep(1); f.close(); time.sleep(1); f=open('config','r'); json.load(f); f.close()
{'dum': 1, 'tests': 7, 'oledd': 0, 'dormant': 1, 'web': 1}
```

but when boot.py runs the json.load retrieves the old parameters from before the latest json.dump. The only way I can reliably get the new values is to run the config write at least 3 times before using boot.py.
 
There's no problem with the pico's flash, the file read/writes in boot.py work OK so I'm wondering what I'm missing?

**Comment by jimmo** [comment]
> before running boot.py

@kjm1102  What do you mean by "before running boot.py".

**Reply by kjm1102** [reply]
I execute it from the same repl I used to store the new values in the config file in flash

**Reply by jimmo** [reply]
How do you "run boot.py" then?

**Comment by scruss** [comment]
How are you viewing the files on the Pico? For me it writes first time, but Thonny's files view has to be manually refreshed to see it

**Reply by kjm1102** [reply]
Actually only need to write the config file once in Thonny for boot.py to see the new values when I run it, so having to do it 3 times must be an Atom problem. Yes, yes, I know Atom has been abandoned but it was my first repl & I'm very comfortable with it.

**Reply by scruss** [reply]
This doesn't sound like a MicroPython issue, rather an Atom sync problem

**Reply by kjm1102** [reply]
I'm dreading getting into Visual studio as a repl but Thonny is awkward with .mpy files & Atom's cranky little glitches like this are gonna force me I fear!
